taskmanager/src/Infrastructure/Adapters/Incoming/JSON/AddTask.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Infrastructure\Adapters\Incoming\JSON;
+
+use App\Domain\Ports\DTO\NewTask;
+use App\Domain\TaskService;
+
+class AddTask implements \App\Domain\Ports\Incoming\AddTask
```

Comment:
IMHO this port & adapter might be useless because such thing should be done by framework layer. E.G. based on the headers you can decode the data and put them to the request

Reply:
Im not sure. We can't skip layer Ports/Adapters and working directly on Domain objects we can discus about this. Sorry if I didn't understand you well.

Reply:
We want to strongly separate framework from domain layer.